Automattic/wp-calypso
# A4A: Modifying SectionNav NavTabs component to consider navigation using horizontal scroll

Related to #https://github.com/Automattic/automattic-for-agencies-dev/issues/123

## Proposed Changes

* This PR modifies the SectionNav NavTabs component to allow an additional prop - `hasHorizontalScroll` - which helps prevent the dropdown being displayed only when the navigation tabs width is greater than the width of it's parent section-nav-group. This makes sense if horizontal scroll for the navigation isn't enabled, but not otherwise.
* Horizontal scroll for the preview pane section nav wasn't currently enabled though, but is now. With the ever increasing amount of navigation tabs this makes sense imo.
* In doing so, I've added a max width which is the generic mobile max width. By adding this, the other issue with the dropdown showing unnecessarily is also prevented: https://github.com/Automattic/automattic-for-agencies-dev/issues/244.
* That said, we could add another prop to set that width. I'm wary about modifying the SectionNav components too much though.

## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

* Open the preview pane in different screen widths, and the dropdown should now only display on mobile (or on screens smaller than 480px).
* You should see horizontal scroll for the navigation menu now as well, where it otherwise would be chopped off.
* Note when modifying the screen size, consider whether that screen size would show the preview pane in it's current state if opening the pane (or opening the page via URL using a URL that would ordinarily open the pane in full screen). Sometimes just manually  changing the width results in a display which wouldn't happen normally at that screen size.

## Pre-merge Checklist
